presto-orc/src/main/java/com/facebook/presto/orc/Filters.java

```diff
@@ -0,0 +1,924 @@
+
+        private final long[] originalValues;
+        private final long[] longs;
+        private final int size;
+        private boolean containsEmptyMarker;
+
+        public BigintValues(long[] values, boolean nullAllowed)
+        {
+            super(nullAllowed);
+
+            requireNonNull(values, "values is null");
+            checkArgument(values.length > 1, "values must contain at least 2 entries");
+
+            this.originalValues = values;
+            this.size = Integer.highestOneBit(values.length * 3);
+            this.longs = new long[size];
+            Arrays.fill(longs, EMPTY_MARKER);
+            for (long value : values) {
+                if (value == EMPTY_MARKER) {
+                    containsEmptyMarker = true;
+                }
+                else {
+                    int pos = (int) ((value * M) & (size - 1));
+                    for (int i = pos; i < pos + size; i++) {
+                        int idx = i & (size - 1);
+                        if (longs[idx] == EMPTY_MARKER) {
+                            longs[idx] = value;
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+
+        @Override
+        public boolean testLong(long value)
+        {
+            if (containsEmptyMarker && value == EMPTY_MARKER) {
+                return true;
+            }
+            int pos = (int) ((value * M) & (size - 1));
+            for (int i = pos; i < pos + size; i++) {
+                int idx = i & (size - 1);
+                long l = longs[idx];
+                if (l == EMPTY_MARKER) {
+                    return false;
+                }
+                if (l == value) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        @Override
+        public boolean equals(Object o)
+        {
+            if (this == o) {
+                return true;
+            }
+
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+
+            BigintValues that = (BigintValues) o;
+            return size == that.size &&
+                    containsEmptyMarker == that.containsEmptyMarker &&
+                    Arrays.equals(longs, that.longs) &&
+                    nullAllowed == that.nullAllowed;
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(size, containsEmptyMarker, longs, nullAllowed);
+        }
+
+        @Override
+        public String toString()
+        {
+            return toStringHelper(this)
+                    .add("values", originalValues)
+                    .add("nullAllowed", nullAllowed)
+                    .toString();
+        }
+    }
+
+    public static class DoubleRange
+            extends Filter
+    {
+        private final double lower;
+        private final boolean lowerUnbounded;
```

Comment:
Would an abstract `RangeFilter` (possibly with a template if performance allows) to avoid repetition of `lowerUnbounded`, `lowerExclusive`, etc in all these classes?